### ISSUE TITLE
Uninstalled/overwritten extension versions running even after restart

### ISSUE THREAD

**Issue by DuncanWalter** [microsoft/vscode#146208]
Does this issue occur when all extensions are disabled?: Yes/No

Not applicable, the issue involves loading non-built-in extensions

- VS Code Version: 1.65.2
- OS Version: mac OS 11.6

Steps to Reproduce:

I suspect the extension host assumes npm semver versions are changing between extension versions and that means some cache isn't evicted properly in cases where npm isn't the package manager being used to distribute/build extensions. 

1. Install a non-bundled extension from VISX
2. Open VS Code and make sure the extension activates
3. Rename a source file in a local dev copy of the extension and make a new VISX (using the same npm semver may be necessary?)
4. Install that new VISX using `--force`
5. Reload VS Code and try to activate the extension. Sometimes, the extension fails to start because the previous version of the extension is running, but it's unable to load relatively imported files that changed names.

**Comment by sandy081** [comment]
Can you please confirm after following step, extracted VSIX in the disk has new files - You can go to extracted location using the command `Open Extensions Folder`?

`Install that new VISX using --force`

**Comment by DuncanWalter** [comment]
Yeah, that takes me to the extensions folder where the new version is installed. That's also the directory the runtime error is supposedly reported from. The runtime error refers to code that doesn't exist on disk in the currently installed version, though

**Comment by sandy081** [comment]
Since it is related to runtime, assigning to @alexdima

**Comment by alexdima** [comment]
This might be the extension code caching. Not sure if that gets deleted when using install vsix `--force`.

**Comment by jrieken** [comment]
> It seems like there's some very specific node runtime / caching weirdness going on. Strangely, this persists even after killing and restarting the entire vscode process tree

Can you try `--no-cached-data` when this happens. This will make us ignore the VM cached data
